### PR TITLE
Fixes MySQL 8.0 compatibility with current SqlService, updates Hikari…

### PR DESCRIPTION
**Pull request by ghost**
This is not intended for merging as-is if MariaDB compatibility is to remain a feature of SpongeCommon. The MariaDB connector can co-exist with the MySQL 8.0.x connector if the SqlService implementation is adjusted to allow for options in the JDBC URL. 
 
Info on this: https://mariadb.com/kb/en/library/about-mariadb-connector-j/#having-mariadb-and-mysql-drivers-in-the-same-classpath

The regex would need to be updated for options (e.g. useSSL, disableMariaDbDriver, etc.). The changes in this PR simply allow the current SqlService to connect to a MySQL 8.0.x database without any invasive changes. The driver name has been updated to the proper 8.0 name.

Tested and working properly **for MySQL 8.0.13**. No other tests have been performed.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -67,8 +67,9 @@ dependencies {
     runtime 'org.apache.logging.log4j:log4j-slf4j-impl:2.8.1'
 
     // Database connectors
-    compile 'com.zaxxer:HikariCP:2.6.3'
-    runtime 'org.mariadb.jdbc:mariadb-java-client:2.0.3'
+    compile 'com.zaxxer:HikariCP:3.2.0'
+    //runtime 'org.mariadb.jdbc:mariadb-java-client:2.0.3'
```
Comment (ryantheleach) [inline build.gradle:71]: 1. This removes mariadb support, which would be a breaking change. additionally it's commented.

File: gradle/implementation.gradle [hunk 2]
```diff
@@ -146,7 +146,8 @@ shadowJar ext.shadowConfiguration = {
 
         include dependency('com.zaxxer:HikariCP')
 
-        include dependency('org.mariadb.jdbc:mariadb-java-client')
+        //include dependency('org.mariadb.jdbc:mariadb-java-client')
```
Comment (ryantheleach) [inline gradle/implementation.gradle:149]: See 1.

File: src/main/java/org/spongepowered/common/service/sql/SqlServiceImpl.java [hunk 3]
```diff
@@ -85,8 +85,8 @@
         mySqlProps.setProperty("useConfigs",
                 "maxPerformance"); // Config options based on http://assets.en.oreilly
                 // .com/1/event/21/Connector_J%20Performance%20Gems%20Presentation.pdf
-        build.put("com.mysql.jdbc.Driver", mySqlProps);
-        build.put("org.mariadb.jdbc.Driver", mySqlProps);
+        build.put("com.mysql.cj.jdbc.Driver", mySqlProps);
+        //build.put("org.mariadb.jdbc.Driver", mySqlProps);
```
Comment (ryantheleach) [inline src/main/java/org/spongepowered/common/service/sql/SqlServiceImpl.java:89]: see 1.

File: src/test/java/org/spongepowered/common/service/sql/SqlServiceImplTest.java [hunk 4]
```diff
@@ -40,7 +40,8 @@ public void testMysqlConnectionInfo() throws SQLException {
         assertEquals("zml", subject.getUser());
         assertEquals("totallymypassword", subject.getPassword());
         assertEquals("jdbc:mysql://localhost/sponge", subject.getAuthlessUrl());
-        assertEquals("org.mariadb.jdbc.Driver", subject.getDriverClassName());
+        //assertEquals("org.mariadb.jdbc.Driver", subject.getDriverClassName());
```
Comment (ryantheleach) [inline src/test/java/org/spongepowered/common/service/sql/SqlServiceImplTest.java:43]: see 1.

**Review by ryantheleach — CHANGES_REQUESTED** [review]
Thanks for the Pull Request!

It's appreciated when community members like yourself have put effort by creating a pull request. 

However **If** we need to add mysql explicitly, we need to do so without dropping mariadb support.

The mariadb connector has explicit support for this, and can let mysql connector handle mysql connectors and let mariadb handle mariadb connections.

Whilst researching your issue, yesterday, I came across https://mariadb.com/kb/en/library/about-mariadb-connector-j/#having-mariadb-and-mysql-drivers-in-the-same-classpath

Which has explicit notes on how to have both loaded.

I'd prefer someone more familiar with databases in Java to confirm if this is actually the route we want to take, As I believe with additional testing it may show that all we need to do is update the mariadb connector which we already use, which would cut down on the amount of dependencies needed.

**Comment by ghost** [comment]
New PR #2152 with completed feature impl.

**Pull request closed without merging**